app/src/test/java/com/example/feelgoodinc/services/UserServiceTest.java

```diff
@@ -182,7 +183,75 @@ public void registerUser_Failed() {
 
         //register user
         userServiceMock.registerUser(email, password, signUpCallback);
-  }
+
+    }
+
+    @Test
+    public void changePassword_Successful() {
+        // Mock user inputs
+        String oldPassword = "Password123!";
+        String newPassword = "NewPassword456!";
+
+        // Mock FirebaseAuth and FirebaseUser
+        FirebaseAuth mockedFirebaseAuth = mock(FirebaseAuth.class);
+        FirebaseUser mockedFirebaseUser = mock(FirebaseUser.class);
+
+        when(mockedFirebaseAuth.getCurrentUser()).thenReturn(mockedFirebaseUser);
+        when(mockedFirebaseUser.getEmail()).thenReturn("test@example.com");
+
+        // Set up the UserService with the mocked FirebaseAuth
+        UserService userServiceMock = new UserService();
+        userServiceMock.mAuth = mockedFirebaseAuth;
+
+        // Mock reauthentication Task
+        @SuppressWarnings("unchecked")
```

Comment:
Having a SupressWarning is a bit strange. If you can, add a comment saying why it's needed. Delete it, if not needed.